book/09-git-and-other-scms/sections/client-bzr.asc

```diff
@@ -0,0 +1,102 @@
+==== Git and Bazaar
+
+There are many projects that allow you to use Git as a Bazaar client.
+Here we'll use Felipe Contreras' project that you may find at https://github.com/felipec/git-remote-bzr[].
+To install it, you just have to download the file git-remote-bzr in a folder contained in your `$PATH`:
+[source,console]
+----
+$ wget https://raw.github.com/felipec/git-remote-bzr/master/git-remote-bzr -O ~/bin/git-remote-bzr
+$ chmod +x ~/bin/git-remote-bzr
+----
+
+You also need to have Bazaar installed.
+
+That's all!
+
+===== Create a Git repository from a Bazaar repository
+
+It is simple to use.
+It is enough to clone a Bazaar repository prefixing it by `bzr::`.
+If you worked in a local Bazaar repository and pushed to a remote one, it is better to clone the remote.
+That will make your work simpler; else you must go through your local repository and you may have problems when pushing to the remote if your Bazaar and Git repositories were not up-to-date.
+
+Let's suppose that you worked with a remote repository which is at address `bzr+ssh://developer@mybazaarserver:myproject`.
+Then you must clone it in the following way:
+[source,console]
+----
+$ git clone bzr::bzr+ssh://developer@mybazaarserver:myproject myProject-Git
+$ cd myProject-Git
+----
+
+You should also clean and compact the created Git repository, especially if it is a big one:
+[source,console]
+----
+$ git gc --aggressive
+----
+
+===== Bazaar branches
+
+Bazaar only allows you to clone branches, but a repository may contain several branches, and `git-remote-bzr` can clone both.
+For example, to clone a branch:
+[source,console]
+----
+$ git config remote-bzr.branches 'trunk, xwindow'
```

Comment:
```console
$ git clone bzr::bzr://bzr.savannah.gnu.org/emacs emacs
```

The final "emacs" is optionnal since the new repository will have the name of the last part of the URL.